Buy/GraphClient.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  GraphClient.swift
+//  Buy
+//
+//  Created by Dima Bart on 2017-02-23.
+//  Copyright © 2017 Shopify. All rights reserved.
+//
+
+import Foundation
+
+internal typealias JSON = [String : Any]
+
+private struct Header {
+    static var userAgent     = "User-Agent"
+    static var authorization = "Authorization"
+    static var accept        = "Accept"
+    static var contentType   = "Content-Type"
+}
+
+public class GraphClient {
+    
+    public let session:  URLSession?
+    
+    private let apiURL:  URL
+    private let headers: [String : String]
+	
+    // ----------------------------------
+    //  MARK: - Init -
+    //
+    public init(shopDomain: String, apiKey: String, session: URLSession = URLSession(configuration: URLSessionConfiguration.default)) {
+        
+        let shopURL  = GraphClient.urlFor(shopDomain)
+        self.apiURL  = shopURL.appendingPathComponent("api/graphql")
+        self.session = session
+        self.headers = [
+            Header.userAgent     : Global.userAgent,
+            Header.authorization : "Basic \(GraphClient.tokenFor(apiKey))"
+        ]
+        
+        precondition(!apiKey.isEmpty, "API Key is required to the Buy SDK. You can obtain one by adding a Mobile App channel here: \(shopURL.appendingPathComponent("admin"))")
+    }
+    
+    static func tokenFor(_ apiKey: String) -> String {
+        return apiKey.data(using: .utf8)!.base64EncodedString()
+    }
+    
+    static func urlFor(_ shopDomain: String) -> URL {
+        return URL(string: "https://\(shopDomain)")!
```

Comment:
I think there is better API for this, to prevent needing to concatenate explicitly.